src/main/java/functions/Parser.java

```diff
@@ -0,0 +1,223 @@
+package functions;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import tasks.*;
+
+public class Parser {
+
+    private String input;
+    private TaskList taskList;
+
+    private enum ChatFunction {
+        LIST,
+        MARK,
+        UNMARK,
+        TODO,
+        DEADLINE,
+        EVENT,
+        DELETE,
+        FIND
+    }
+
+    /**
+     * A public constructor to initialize a new Parser instance
+     *
+     * @param input file path to load
+     * @param taskList a task list containing all the tasks stored
+     */
+    public Parser(String input, TaskList taskList) {
+        this.input = input;
+        this.taskList = taskList;
+    }
+
+    public void parse() {
+        String[] inputArray = input.split(" ");
+
+        try {
+            ChatFunction function = ChatFunction.valueOf(inputArray[0].toUpperCase());
+            int firstSpaceIndex = input.indexOf(" ");
+            String secondHalfInput = input.substring(firstSpaceIndex+1);
+
+            switch (function) {
+                case LIST:
+                    listCommandHandler(taskList);
+                    break;
+
+                case MARK:
+                    markCommandHandler(inputArray, taskList);
+                    break;
+
+                case UNMARK:
+                    unmarkCommandHandler(inputArray, taskList);
+                    break;
+
+                case DELETE:
+                    deleteCommandHandler(inputArray, taskList);
+                    break;
+
+                case TODO:
+                    toDoCommandHandler(taskList, secondHalfInput);
+                    break;
+
+                case DEADLINE:
+                    deadlineCommandHandler(taskList, secondHalfInput);
+                    break;
+
+                case EVENT:
+                    eventCommandHandler(taskList, input, secondHalfInput);
+                    break;
+
+                case FIND:
+                    findCommandHandler(taskList, secondHalfInput);
+
+                default:
+                    break;
+            }
```

Comment:
```suggestion
switch (condition) {
case ABC:
    statements;
    // Fallthrough
case DEF:
    statements;
    break;
case XYZ:
    statements;
    break;
default:
    statements;
    break;
}
```

Note there is no indentation for case clauses.